MVP-UI/MVP-UI/RecipesModule/ReccipesList/RecipesListViewController.swift

```diff
@@ -173,22 +187,69 @@ extension RecipesListViewController: UITableViewDataSource {
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        guard let cell = tableView.dequeueReusableCell(
-            withIdentifier: Constants.cellIdendefire,
-            for: indexPath
-        ) as? RecipesCell
-        else { return UITableViewCell() }
-        guard let recipe = recipes?.recepies[indexPath.row] else { return cell }
-        cell.configure(with: recipe)
-        return cell
+        switch stateShimer {
+        case .loading:
+            return ShimerViewCell()
+        case .done:
+            guard let cell = tableView.dequeueReusableCell(
+                withIdentifier: Constants.cellIdendefire,
+                for: indexPath
+            ) as? RecipesCell
+            else { return UITableViewCell() }
+            guard let recipe = recipes?.recepies[indexPath.row] else { return cell }
+            cell.configure(with: recipe)
+            return cell
+        }
     }
 }
 
 // MARK: - Extension + RecipesViewProtocol
 
 extension RecipesListViewController: RecipesViewProtocol {
+    func changeShimerState() {
+        DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
+            self.stateShimer = .done
+            self.recipesTableView.reloadData()
+        }
+    }
+
+    func buttonCaloriesState(color: String, image: String) {
+        caloriesButton.backgroundColor = UIColor(named: color)
+        caloriesButton.setTitleColor(.white, for: .normal)
+        caloriesButton.setImage(UIImage(named: image), for: .normal)
+        caloriesButton.setTitleColor(.black, for: .normal)
```

Comment:
Один метод на настройку всей кнопки в зависимости от переденного в нее стейта
```suggestion
        caloriesButton.setButton(with style: ТутТвой ЭнамСтейтов, imageName: String)
```
а а цвета и прочее устанавливаются уже в зависимости от стейта
из Презентера удаляем import UIKit
он UI независимый